### PR TITLE
Fixed uploading by using Modal

### LINKED ISSUE #11: Error: any ideas?
Hey Thanks for doing this.  Very cool and easy.  

it was working fine until today.  i did pull your latest release.  getting this error today using your sample code with a valid file location. on Windows 10 and FF 86.0 (64-bit)

PS C:\Users\johnj> & python c:/Users/johnj/Desktop/auto-downloader/TestUpload03152021.py
DEBUG: Firefox is now running
DEBUG: Trying to upload "D:/ToYouTube/upload/365tour.mp4" to YouTube...
DEBUG: Trying to set "My YouTube Title" as title...
DEBUG: Trying to set "My YouTube Description" as description...
Traceback (most recent call last):
  File "c:/Users/johnj/Desktop/auto-downloader/TestUpload03152021.py", line 8, in <module>
    was_uploaded, video_id = upload.upload(
  File "C:\Users\johnj\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.8_qbz5n2kfra8p0\LocalCache\local-packages\Python38\site-packages\opplast\upload.py", line 78, in upload        
    container = self.driver.find_element_by_xpath(DESCRIPTION_CONTAINER)
  File "C:\Users\johnj\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.8_qbz5n2kfra8p0\LocalCache\local-packages\Python38\site-packages\selenium\webdriver\remote\webdriver.py", line 
394, in find_element_by_xpath
    return self.find_element(by=By.XPATH, value=xpath)
  File "C:\Users\johnj\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.8_qbz5n2kfra8p0\LocalCache\local-packages\Python38\site-packages\selenium\webdriver\remote\webdriver.py", line 
976, in find_element
    return self.execute(Command.FIND_ELEMENT, {
  File "C:\Users\johnj\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.8_qbz5n2kfra8p0\LocalCache\local-packages\Python38\site-packages\selenium\webdriver\remote\webdriver.py", line 
321, in execute
    self.error_handler.check_response(response)
  File "C:\Users\johnj\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.8_qbz5n2kfra8p0\LocalCache\local-packages\Python38\site-packages\selenium\webdriver\remote\errorhandler.py", line 242, in check_response
    raise exception_class(message, screen, stacktrace)
selenium.common.exceptions.NoSuchElementException: Message: Unable to locate element: /html/body/ytcp-uploads-dialog/tp-yt-paper-dialog/div/ytcp-animatable[1]/ytcp-uploads-details/div/ytcp-uploads-basics/ytcp-mention-textbox[2]

### PR DESCRIPTION
**Pull request by breuerfelix**
hey, fixed some stuff (i was also experiencing issue #11)

- search for the 'upload dialog modal' and make all other searches according to that
- removed one next-button since you have to press only twice
- text.clear() does not clear. spammed backspace which works 100%
- used crlf as line endings since you are using windows
- fixed typing errors

about the line endings...  
https://stackoverflow.com/questions/1889559/git-diff-to-ignore-m -> you should change your git settings to "Checkout Windows-style, commit Unix-style line endings" since this is state of the art and you will run into problems when making changes to other people repositories because you will always commit wrong line endings.  
once you made the switch -> remove .gitattributes file :)

fixes #11

### CODE REVIEW

File: opplast/upload.py [hunk 1]
```diff
@@ -126,21 +135,18 @@ def upload(self, meta: dict) -> (bool, str):
                     "Did not set tags. Tags cannot be longer than 500 characters"
                 )
 
-        self.driver.find_element_by_id(NEXT_BUTTON).click()
```
Comment (offish) [inline opplast/upload.py:129]: Why did you delete the 3rd "next" click? When I upload, I get 3 tabs. By only clicking next twice I would run into an exception, because it can't find the public button.
image

Comment (breuerfelix) [reply opplast/upload.py:129]: this is pretty weird since i only got 3 tabs 
image

maybe we should wrap the thrid next button around try catch to solve this :)

Comment (breuerfelix) [reply opplast/upload.py:129]: fixed it

File: opplast/upload.py [hunk 2]
```diff
@@ -52,19 +54,26 @@ def upload(self, meta: dict) -> (bool, str):
...
                 self.log.debug(f'Trying to set "{title}" as title...')
-                title_field = self.driver.find_element_by_id(TEXTBOX)
+                title_field = modal.find_element_by_id(TEXTBOX)
                 title_field.click()
                 sleep(self.timeout)
 
-                title_field.clear()
-                sleep(self.timeout)
+                # clearing out title which defaults to filename
+                for i in range(30):
```
Comment (offish) [inline opplast/upload.py:73]: Wouldn't this be better?
```python
 for i in range(len(title_field.text)): 
``` 
If the file name is longer than 30 characters, it wouldn't get completely cleared.

Comment (breuerfelix) [reply opplast/upload.py]: i added a sleep delay and +10 because sometime a backspace might be missing :) just in case!

**Comment by offish** [comment]
I didn't completely understand the crlf/line ending thing, but I'll look into it! Thank you for this PR.

**Comment by breuerfelix** [comment]
well windows uses other characters for line endings than linux. in linux `\n` represents a newline and in windows this is some other character. when installing git on windows you can choose the option to convert all your line endings to unix line endings when commit and on your pc you have windows line endings.  
yeah a little confusing. this doesnt matter as long as you only work with yourself or with other people using windows :) but in general all repositories use unix line endings

thanks for merging maybe we could also implement the feature requests i filed in the issues :)

**Comment by offish** [comment]
Ah okay, got it, thanks. Yeah, these features would be a nice addition to the repo and make it better. I'll definitely try to work on/merge these.

**Comment by breuerfelix** [comment]
> Ah okay, got it, thanks. Yeah, these features would be a nice addition to the repo and make it better. I'll definitely try to work on/merge these.

awesome! once i got some quota again i might pick one up aswell :)
